Mediator.h

```diff
@@ -10,13 +10,12 @@ template <typename Item> class Mediator
 {
 public:
   
-  Mediator( int nItems )
+  Mediator(int nItems ):N(nItems)
```

Comment:
It is an initialization list. when you call a constructor it takes care of the instantiation of all class member before executing the code itself. Using an initialization list allow to specify how to instatiate those members by passing arguments to their constructors. This avoids instatiation with defaults just for reallocating/change them in constructor body.
In this case, for instance the original code is equivalent to: int N; N = nItems;   the new instead to int N = nItems;
The overhead is almost none for ints but maybe huge for some custom class. 
Also the initialization list is where you call constructors of an hypotetical base class.


if you would like to have more details see for instance:
http://www.learncpp.com/cpp-tutorial/101-constructor-initialization-lists/